pkg/apis/metal/types_controlplane.go

```diff
@@ -21,10 +21,19 @@ type ControlPlaneConfig struct {
 	LoadBalancerConfig *LoadBalancerConfig
 }
 
+// CloudControllerNetworking contains configuration settings for CCM networking.
+type CloudControllerNetworking struct {
+	// ConfigureNodeAddresses enables the configuration of node addresses.
+	ConfigureNodeAddresses bool
+}
+
 // CloudControllerManagerConfig contains configuration settings for the cloud-controller-manager.
 type CloudControllerManagerConfig struct {
 	// FeatureGates contains information about enabled feature gates.
 	FeatureGates map[string]bool
+
+	// CloudControllerNetworking contains configuration settings for CCM networking.
+	CloudControllerNetworking *CloudControllerNetworking
```

Comment:
Shouldn't we just call this field `Networking`.

Reply:
@afritzler I am not sure taking the field name `Networking` is the right thing. Because there may be other types related to networking in other controlplane components e.g. in LB config.

Reply:
You can keep the `*CloudControllerNetworking` as the type but `CloudControllerNetworking` as the field name. That way the reference will be: `CloudControllerManagerConfig.Networking` instead of `CloudControllerManagerConfig.CloudControllerNetworking`.

Reply:
that makes perfect sense  - I have changed it